### ISSUE TITLE
Frame position logged as 0 both when no pattern is displayed and when the first frame of a pattern is displayed - Priority 6

### ISSUE THREAD

**Issue by taylorlm88** [JaneliaSciComp/G4_Display_Tools#49]
Summary: Log frame position as ‘NaN’ when nothing is shown on the display.

Description: The Frames_Position log file currently provides the frame number of the current pattern that is being displayed on the screen at any given time. However, we have noticed that between patterns, when nothing is being displayed, the position number given is 0. When the first frame of a pattern is being displayed, the position number given is also 0. It would be beneficial to us to be able to distinguish these two states, if nothing else so that we can determine exactly when the pattern first displays on the screen. Would it be possible to change the frame position logged when nothing is being displayed on the screen to a NaN or some very large number that would never occur in this data otherwise? (Like 1000).

**Comment by taylorlm88** [comment]
I don't believe this has been done. In a trial with a 192 frame pattern, the numbers shown in Frames.Position range from 0 to 191, meaning 0 must represent the first frame. I don't see anything at the beginning other than 0s to represent no frame being displayed at all. To be extra sure, I will do a test on the arena where I send the commands manually and make sure I log a good amount of time before sending the first pattern to display. That way I can know for sure where the pattern should start displaying and confirm what's in the Frames.Position array. 

That said, I think we may have decided this was unnecessary. We can calculate the time when the pattern first started moving without this update - did we decide this is good enough?

**Comment by taylorlm88** [comment]
Actually, it turns out this HAS been done! I ran a test on the arena in which the log was started and I then paused the code before  running the first condition. I added another pause between the first and second condition. The intention was to be able to see what was being logged under Frame position when nothing was displayed compared to when frame 1 was displayed. It turns out the log did not log any data at all during the pauses when nothing was being displayed.  Even though the log was running for a total of 16.2 seconds (two 6 second pauses and 2 2.1 second conditions), only 4.2 seconds (the conditions) was in the log. The 6 second pause between the two conditions lined up with a large 6 second gap in the timestamps between one condition and the next. 

This is good to know, since in the past we've had issues where a gap in the timestamps caused errors with the data processing. It could be that the log is also not collecting and skipping time if the screen is not displaying by error. I'm not sure, right now I only know that it's not logging when nothing has been sent to the screen. 

This could be used to determine when the pattern first appears on the screen in some but not all cases. If the previous pattern ends on frame 1 and the next one starts on frame 1, there is still no sure way to tell when the transition happens unless there's a gap in the timestamps. I can test using a full protocol to see if there is always a tell tale gap in the timestamps or not. Either way, closing this issue, as it has  been addressed.
